Sources/Sentry/SentryBreadcrumbTracker.m

```diff
@@ -26,16 +26,52 @@ - (void)trackApplicationUIKitNotifications {
-                                                    crumb.message = @"Memory Warning";
-                                                    [SentrySDK addBreadcrumb:crumb];
+                                                    [self addBreadcrumbWithType: @"system"
+                                                                   withCategory: @"device"
+                                                                      withLevel: kSentryLevelWarning
+                                                                    withDataKey: @"action"
+                                                                  withDataValue: @"UIApplicationDidReceiveMemoryWarningNotification"];
+                                                }];
+    
+    [NSNotificationCenter.defaultCenter addObserverForName:UIApplicationDidEnterBackgroundNotification
+                                                    object:nil
+                                                     queue:nil
+                                                usingBlock:^(NSNotification *notification) {
+                                                    [self addBreadcrumbWithType: @"navigation"
+                                                                   withCategory: @"app.lifecycle"
+                                                                      withLevel: kSentryLevelInfo
+                                                                    withDataKey: @"state"
+                                                                  withDataValue: @"background"];
+                                                }];
+    
+    [NSNotificationCenter.defaultCenter addObserverForName:UIApplicationDidBecomeActiveNotification
+                                                    object:nil
+                                                     queue:nil
+                                                usingBlock:^(NSNotification *notification) {
+                                                    [self addBreadcrumbWithType: @"navigation"
+                                                                   withCategory: @"app.lifecycle"
+                                                                      withLevel: kSentryLevelInfo
+                                                                    withDataKey: @"state"
+                                                                  withDataValue: @"foreground"];
                                                 }];
 #else
     [SentryLog logWithMessage:@"NO UIKit -> [SentryBreadcrumbTracker trackApplicationUIKitNotifications] does nothing." andLevel:kSentryLogLevelDebug];
 #endif
 }
-     
+
+- (void)addBreadcrumbWithType:(NSString *)type
+                 withCategory:(NSString *)category
+                    withLevel:(SentryLevel *)level
+                  withDataKey:(NSString *)key
+                withDataValue:(NSString *)value {
+    if (nil != [SentrySDK.currentHub getClient]) {
```

Comment:
m: Why do we have this check? If it's just an optimization, since we're anyway using the static API below, I suggest we don't optimize this here and let the `addBreadcrumb` in the static API do that instead.

Reply:
@bruno-garcia this already exist for all the other breadcrumbs, I just kept the pattern.
I can remove it but it will be different than all the others.
@HazAT is there a reason?

Reply:
It's worth noting this is racey since after the check, a `nil` client could have been bound to `SentrySDK` anyway so IMO this check doesn't make sense